pro/src/ui-kit/form/MultiSelectAutoComplete/__specs__/MultiSelectAutocomplete.spec.tsx

```diff
@@ -11,234 +11,238 @@ import { Formik } from 'formik'
-          screen.getByRole('button', { name: 'Calvados' })
-        ).toBeInTheDocument()
-      })
-
-      it('should not display associated tags when the user selects options', async () => {
-        // given
-        render(
-          <Formik initialValues={initialValues} onSubmit={() => {}}>
-            <MultiSelectAutocomplete {...props} hideTags />
-          </Formik>
-        )
-        await userEvent.click(screen.getByRole('textbox'))
-        await userEvent.click(await screen.findByLabelText('Aveyron'))
-        await userEvent.click(await screen.findByLabelText('Calvados'))
-        expect(
-          screen.queryByRole('button', { name: 'Aveyron' })
-        ).not.toBeInTheDocument()
-        expect(
-          screen.queryByRole('button', { name: 'Calvados' })
-        ).not.toBeInTheDocument()
-      })
-
-      it('should unselect option when removing a tag', async () => {
-        render(
-          <Formik initialValues={initialValues} onSubmit={() => {}}>
-            <MultiSelectAutocomplete {...props} />
-          </Formik>
-        )
-
-        await userEvent.click(screen.getByRole('button', { name: 'Aisne' }))
-        expect(
-          screen.queryByRole('button', { name: 'Aisne' })
-        ).not.toBeInTheDocument()
-        await userEvent.click(screen.getByRole('textbox'))
-        expect(screen.queryByLabelText('Aisne')).not.toBeChecked()
-      })
+    it('should display options as selected when the user selects them', async () => {
+      render(
+        <Formik initialValues={initialValues} onSubmit={jest.fn()}>
+          <MultiSelectAutocomplete {...props} />
+        </Formik>
+      )
+      await userEvent.click(screen.getByRole('textbox'))
+      await userEvent.click(await screen.findByLabelText('Aveyron'))
+      await userEvent.click(await screen.findByLabelText('Calvados'))
+      expect(screen.getAllByRole('checkbox', { checked: true })).toHaveLength(
+        initialValues.departement.length + 2
+      )
+    })
+
+    it('should not display options as selected when the user unselects them', async () => {
+      render(
+        <Formik initialValues={initialValues} onSubmit={jest.fn()}>
+          <MultiSelectAutocomplete {...props} />
+        </Formik>
+      )
+      await userEvent.click(screen.getByRole('textbox'))
+      await userEvent.click(await screen.findByLabelText('Ain'))
+      expect(screen.getAllByRole('checkbox', { checked: true })).toHaveLength(
+        initialValues.departement.length - 1
+      )
+    })
+  })
+
+  describe('filter', () => {
+    it('should filter options when the user types in the field', async () => {
+      render(
+        <Formik initialValues={initialValues} onSubmit={jest.fn()}>
+          <MultiSelectAutocomplete {...props} />
+        </Formik>
+      )
+      await userEvent.type(screen.getByRole('textbox'), 'al')
+      expect(screen.getAllByRole('checkbox')).toHaveLength(6) // Allier, Alpes, Hautes-Alpes, Alpes-Maritimes, Calvados, Cantal
```

Comment:
j'aime ni les magics numbers ni les commentaires qui sont là pour expliciter du code qui est trop implicite

que penses-tu de ?

```suggestion
      expect(screen.getAllByRole('checkbox')).toHaveLength(['Allier', 'Alpes', 'Hautes-Alpes', 'Alpes-Maritimes', 'Calvados', 'Cantal'].length)
```